docs/core/diagnostics/metrics-collection.md

```diff
@@ -0,0 +1,404 @@
+
+#### Configure the sample application to use OpenTelemetry's Prometheus exporter
+
+1. Add a reference to the OpenTelemetry Prometheus exporter to the sample application:
+
+```dotnetcli
+dotnet add package OpenTelemetry.Exporter.Prometheus --version 1.2.0-beta1
+```
+> [!NOTE]
+> The Promethus exporter library includes a reference to OpenTelemetry's shared library so this command implicitly adds both libraries
+> to the application.
+
+2. Modify the code of Program.cs so that it contains the extra code to configure OpenTelemetry at the beginning of Main():
+
+```C#
+using System;
+using System.Diagnostics.Metrics;
+using System.Threading;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+
+class Program
+{
+    static Meter s_meter = new Meter("HatCo.HatStore", "1.0.0");
+    static Counter<int> s_hatsSold = s_meter.CreateCounter<int>(name: "hats-sold", 
+                                                                unit: "Hats",
+                                                                description: "The number of hats sold in our store"); 
+
+    static void Main(string[] args)
+    {
+        using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter("HatCo.HatStore")
+                .AddPrometheusExporter(opt =>
+                {
+                    opt.StartHttpListener = true;
+                    opt.HttpListenerPrefixes = new string[] { $"http://localhost:9184/" };
+                })
+                .Build();
+
+        Console.WriteLine("Press any key to exit");
+        while(!Console.KeyAvailable)
+        {
+            // Pretend our store has a transaction each second that sells 4 hats
+            Thread.Sleep(1000);
+            s_hatsSold.Add(4);
+        }
+    }
+}
+```
+
+`AddMeter("HatCo.HatStore")` configures OpenTelemetry to transmit all the metrics collected by the Meter our app defined.
+`AddPrometheusExporter(...)` configures OpenTelemetry to expose Prometheus' metrics endpoint on port 9184 and to use
+the HttpListener. See the [OpenTelemetry documentation](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.Prometheus)
+for more information about OpenTelemetry configuration options, in particular alternate hosting options that are useful for ASP.NET applications.
+
+> [!NOTE]
+> At the time of writing OpenTelemetry only supports metrics emitted using the <xref:System.Diagnostics.Metrics?displayProperty=nameWithType>
+> APIs however support for [EventCounters](event-counters.md) is planned.
+
+3. Run the sample app and leave it running in the background.
+
+```dotnetcli
+> dotnet run
+Press any key to exit
+```
+
+#### Setup and Configure Prometheus
+
+1. Follow the [Prometheus first steps](https://prometheus.io/docs/introduction/first_steps/) to set up your Prometheus server
+and confirm it is working.
+
+2. Modify the prometheus.yml configuration file so that Prometheus will scrape the metrics endpoint that our example app is
```

Comment:
I noticed that we have both "example" and "sample" used, I suggest that we use "example" (sample/sampling/sampler have special meaning in the overall telemetry space).